core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -669,13 +668,15 @@ public PlanOptimizers(
                         ruleStats,
                         statsCalculator,
                         estimatedExchangesCostCalculator,
-                        ImmutableSet.of(
-                                new RemoveEmptyUnionBranches(),
-                                new EvaluateEmptyIntersect(),
-                                new RemoveEmptyExceptBranches(),
-                                new RemoveRedundantIdentityProjections(),
-                                new PushAggregationThroughOuterJoin(),
-                                new ReplaceRedundantJoinWithSource())), // Run this after PredicatePushDown optimizer as it inlines filter constants
+                        ImmutableSet.<Rule<?>>builder()
+                                .add(new RemoveEmptyUnionBranches())
+                                .add(new EvaluateEmptyIntersect())
+                                .add(new RemoveEmptyExceptBranches())
+                                .add(new RemoveRedundantIdentityProjections())
+                                .add(new PushAggregationThroughOuterJoin())
+                                .add(new ReplaceRedundantJoinWithSource()) // Run this after PredicatePushDown optimizer as it inlines filter constants
+                                .add(new MultipleDistinctAggregationToMarkDistinct()) // Run this after aggregation pushdown
```

Comment:
... so that multiple distinct aggregations can be pushed into a connector

Reply:
i wonder whether we actually considered solving the problem differently. instead of changing rule order, we could have a rule that matches to MarkDistinct and invokes aggregation pushdown.

trying this out could be a follow up. if successful, we could simply this class, presumably even rolling the changes back

Reply:
`MarkDistinct` is not always a result of the `MultipleDistinctAggregationToMarkDistinct` rule. And when it is, it could be hard to identify the original aggregations because of intervening rules.
Also, applying rule `MultipleDistinctAggregationToMarkDistinct` later might be possibly beneficial from the viewpoint of decorrelation.